src/measure_theory/interval_integral.lean

```diff
@@ -385,19 +383,94 @@ begin
+
+lemma integral_comp_div (hc : c ≠ 0) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (x / c) = c • ∫ x in a/c..b/c, f x :=
+by simpa only [inv_inv'] using integral_comp_mul_right (inv_ne_zero hc) hfm
+
+lemma integral_comp_add_right (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (x + d) = ∫ x in a+d..b+d, f x :=
+have A : ae_measurable f (measure.map (λ x, x + d) volume), by rwa [real.map_volume_add_right],
+calc ∫ x in a..b, f (x + d) = ∫ x in a+d..b+d, f x ∂(measure.map (λ x, x + d) volume) :
+  by simp only [interval_integral, set_integral_map measurable_set_Ioc A (measurable_add_right _),
+    preimage_add_const_Ioc, add_sub_cancel]
+... = ∫ x in a+d..b+d, f x : by rw [real.map_volume_add_right]
+
+lemma integral_comp_mul_add (hc : c ≠ 0) (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (c * x + d) = c⁻¹ • ∫ x in c*a+d..c*b+d, f x :=
+by rw [← integral_comp_add_right d hfm, ← integral_comp_mul_left hc $
+  hfm.comp_measurable' (measurable_add_right d) $ by rwa real.map_volume_add_right]
+
+lemma integral_comp_add_mul (hc : c ≠ 0) (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (d + c * x) = c⁻¹ • ∫ x in d+c*a..d+c*b, f x :=
+by simpa only [add_comm] using integral_comp_mul_add hc d hfm
+
+lemma integral_comp_div_add (hc : c ≠ 0) (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (x / c + d) = c • ∫ x in a/c+d..b/c+d, f x :=
+by simpa only [div_eq_inv_mul, inv_inv'] using integral_comp_mul_add (inv_ne_zero hc) d hfm
+
+lemma integral_comp_add_div (hc : c ≠ 0) (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (d + x / c) = c • ∫ x in d+a/c..d+b/c, f x :=
+by simpa only [div_eq_inv_mul, inv_inv'] using integral_comp_add_mul (inv_ne_zero hc) d hfm
+
+lemma integral_comp_mul_sub (hc : c ≠ 0) (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (c * x - d) = c⁻¹ • ∫ x in c*a-d..c*b-d, f x :=
+by simpa only [sub_eq_add_neg] using integral_comp_mul_add hc (-d) hfm
+
+lemma integral_comp_sub_mul (hc : c ≠ 0) (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (d - c * x) = c⁻¹ • ∫ x in d-c*b..d-c*a, f x :=
+begin
+  simp only [sub_eq_add_neg, neg_mul_eq_neg_mul],
+  rw [integral_comp_add_mul (neg_ne_zero.mpr hc) d hfm, integral_symm],
+  simp only [inv_neg, smul_neg, neg_neg, neg_smul],
+end
+
+lemma integral_comp_div_sub (hc : c ≠ 0) (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (x / c - d) = c • ∫ x in a/c-d..b/c-d, f x :=
+by simpa only [div_eq_inv_mul, inv_inv'] using integral_comp_mul_sub (inv_ne_zero hc) d hfm
+
+lemma integral_comp_sub_div (hc : c ≠ 0) (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (d - x / c) = c • ∫ x in d-b/c..d-a/c, f x :=
+by simpa only [div_eq_inv_mul, inv_inv'] using integral_comp_sub_mul (inv_ne_zero hc) d hfm
+
+lemma integral_comp_sub_right (d : ℝ) (hfm : ae_measurable f) :
+  ∫ x in a..b, f (x - d) = ∫ x in a-d..b-d, f x :=
+by simpa only [sub_eq_add_neg] using integral_comp_add_right (-d) hfm
+
+lemma integral_comp_sub_left (d : ℝ) (hfm : ae_measurable f) :
```

Comment:
In this lemma, and in all the other ones, I don't think you need the almost-everywhere measurability assumption: one function you integrate will be almost everywhere measurable if and only if the other one also is.

Reply:
@sgouezel Can you please elaborate? I'm not sure I understand what you mean. 

I needed to include the ae measurability assumption in these lemmas because `integral_comp_mul_right`, `integral_comp_neg`, and `integral_comp_add_right` all included that same assumption. I did not write those three lemmas, so I had assumed that the assumption was necessary for their respective proofs. Are you saying that is not the case?

Reply:
Yes, they should not be necessary in these lemmas. If you don't know how to do that, I can have a go at removing them.

Reply:
Hmm...give me some time to think about it and if I can't figure it out I'll let you know.

Reply:
I have removed them. The only issue is that now I can't review the PR any more as I have added a fair amount of material, so you will need another maintainer to merge it :-)